docker-app/qfieldcloud/core/management/commands/extracts3data.py

```diff
@@ -0,0 +1,113 @@
+import csv
+import logging
+from sys import stdout
+from typing import Generator, NamedTuple
+
+import boto3
+import mypy_boto3_s3
+from django.core.management.base import BaseCommand, CommandParser
+
+logger = logging.getLogger(__name__)
+
+
+class S3ConfigObject(NamedTuple):
+    storage_access_key_id: str
+    storage_secret_access_key: str
+    storage_bucket_name: str
+    storage_endpoint_url: str
+    storage_region_name: str
+
+
+class Command(BaseCommand):
+    """Save metadata from S3 storage project files to disk."""
+
+    def add_arguments(self, parser: CommandParser):
+        parser.add_argument(
+            "-o", "--output", type=str, help="Optional: Name of output file"
+        )
+        parser.add_argument(
+            "--storage_access_key_id",
+            type=str,
+            help="S3 access key ID",
+        )
+        parser.add_argument(
+            "--storage_secret_access_key",
+            type=str,
+            help="S3 access key secret",
+        )
+        parser.add_argument(
+            "--storage_bucket_name",
+            type=str,
+            help="S3 bucket name",
+        )
+        parser.add_argument(
+            "--storage_endpoint_url",
+            type=str,
+            help="S3 endpoint url",
+        )
+        parser.add_argument(
+            "--storage_region_name",
+            type=str,
+            help="S3 region name",
+        )
+
+    def handle(self, **options):
+        config = S3ConfigObject(
+            **{
```

Comment:
Avoid using dictionary comprehensions that take more than one line or even avoid them at all. THey are hard to read for people who are changing languages all the time.